extension/encoding/awscloudwatchmetricstreamsencodingextension/json_unmarshaler.go

```diff
@@ -139,20 +140,10 @@ func (c *formatJSONUnmarshaler) UnmarshalMetrics(record []byte) (pmetric.Metrics
 	for datumIndex := 0; scanner.Scan(); datumIndex++ {
 		var cwMetric cloudwatchMetric
 		if err := jsoniter.ConfigFastest.Unmarshal(scanner.Bytes(), &cwMetric); err != nil {
-			c.logger.Error(
-				"Unable to unmarshal input",
-				zap.Error(err),
-				zap.Int("datum_index", datumIndex),
-			)
-			continue
+			return pmetric.Metrics{}, fmt.Errorf("error unmarshaling datum at index %d: %w", datumIndex, err)
```

Comment:
Do we have a test for this branch? I guess not, as I don't see any tests updated with this error message.

Reply:
Added.